koku/api/settings/tags/mapping/query_handler.py

```diff
@@ -25,3 +35,24 @@ def format_tag_mapping_relationship(original_response):
     formatted_response = Response(formatted_data)
 
     return formatted_response
+
+
+@dataclass
+class AddChildQueryHandler:
+    serializer: AddChildSerializer
+    children_rows: List[EnabledTagKeys] = field(default_factory=list)
+    parent_row: EnabledTagKeys = field(init=False)
+
+    def __post_init__(self):
+        """Collects the enabled tag keys rows for the parent & children."""
+        self.parent_row = EnabledTagKeys.objects.get(uuid=self.serializer.data.get("parent"))
+        self.children_rows = list(EnabledTagKeys.objects.filter(uuid__in=self.serializer.data.get("children")))
```

Comment:
What do we gain by using `dataclass` here?

Reply:
Long story short, this used to be combined with the other data class and I decided to pull the FindTagKeyProviders out at the time. Thanks for pointing it out, I think I found a cleaner alternative.